staging/src/k8s.io/client-go/tools/leaderelection/leaderelection_test.go

```diff
@@ -917,3 +917,180 @@ func TestTryAcquireOrRenewEndpointsLeases(t *testing.T) {
 func TestTryAcquireOrRenewConfigMapsLeases(t *testing.T) {
 	testTryAcquireOrRenewMultiLock(t, "configmapsleases")
 }
+func TestTryToReleaseEndpointsLeases(t *testing.T) {
+	testTryReleaseMultiLock(t, "endpointsleases")
+}
+func TestTryReleaseConfigMapsLeases(t *testing.T) {
+	testTryReleaseMultiLock(t, "configmapsleases")
+}
+func testTryReleaseMultiLock(t *testing.T, objectType string) {
+	future := time.Now().Add(1000 * time.Hour)
+	primaryType, secondaryType := multiLockType(t, objectType)
+	tests := []struct {
+		name              string
+		observedRecord    rl.LeaderElectionRecord
+		observedRawRecord []byte
+		observedTime      time.Time
+		reactors          []Reactor
+		releaseReactors   []Reactor
+
+		expectSuccess    bool
+		transitionLeader bool
+		doRelease        bool
+		outHolder        string
+	}{
+		{
```

Comment:
add test cases for the other scenarios please (thank you for adding a test)

Reply:
sorry, which other scenarios? the test included is triggered in case of the original issue (LeaseDurationSeconds not being set) which in real life situation is triggered by the server validation. I can add other types of locking mechanism, but since we are faking validation through reactors it wont have a lot of value imho.

Reply:
`acquire and lease from object without defaults set`, `acquire and lease from object with defaults set`, `acquire and lease from object with non-default values set` are all scenarios i'd expect to see